packages/firestore/src/core/view.ts

```diff
@@ -301,9 +304,15 @@ export class View {
 
     this.applyTargetChange(targetChange);
     const limboChanges = updateLimboDocuments
-      ? this.updateLimboDocuments()
+      ? this.updateLimboDocuments(waitForRequeryResult ?? false)
```

Comment:
Optional: Consider putting `waitForRequeryResult = waitForRequeryResult ?? false;` near the top of this function. IMO this improves readability later on because you then don't have to think about how `undefined` will be coerced to a boolean on lines 307 and 314.